dagger/main.go

```diff
@@ -0,0 +1,42 @@
+// A generated module for Dagger functions
+//
+// This module has been generated via dagger init and serves as a reference to
+// basic module structure as you get started with Dagger.
+//
+// Two functions have been pre-created. You can modify, delete, or add to them,
+// as needed. They demonstrate usage of arguments and return types using simple
+// echo and grep commands. The functions can be called from the dagger CLI or
+// from one of the SDKs.
+//
+// The first line in this comment block is a short description line and the
+// rest is a long description with more detail on the module's purpose or usage,
+// if appropriate. All modules should have a short description.
+
+package main
+
+import (
+	"context"
+
+	"dagger/dagger/internal/dagger"
+	"fmt"
+)
+
+type Dagger struct{}
+
+func (m *Dagger) TestChain(ctx context.Context, chain string, foundAmount string, transferedAmount string, toAddress string, image string, source *dagger.Directory) (string, error) {
+	nodeService := dag.Container().
+		From(image).
+		WithExposedPort(10000).
+		WithExposedPort(10001).
+		AsService()
+
+	privateKey := "22194a8955e9233aa2f0a0206c8ea861e5fa92a613ab5c7e236a11de3f4bc9ad"
+	return dag.Container().
+		From("us-docker.pkg.dev/cordialsys/crosschain/rockylinux").
+		WithDirectory("/crosschain", source).
+		WithServiceBinding("node-service", nodeService).
+		WithEnvVariable("PRIVATE_KEY", privateKey).
+		WithExec([]string{"sh", "-c", fmt.Sprintf("cd crosschain/ci/ && go test -run TestBalance --chain %s --foundAmount %s --rpc http://node-service:10000", chain, foundAmount)}).
```

Comment:
Regarding excluding these tests from running as unit tests.

Should add build tag `ci` to crosschain/ci/ test files.

```
//go:build ci
```

Then when running from dagger, we should run:

```bash
go test -tags ci <... other args>
```